pkg/storage/unified/apistore/store.go

```diff
@@ -529,64 +539,32 @@ func (s *Storage) GuaranteedUpdate(
 		break
 	}
 
-	unchanged, err := isUnchanged(s.codec, existingObj, updatedObj)
+	req.Value, err = s.prepareObjectForUpdate(ctx, updatedObj, existingObj)
 	if err != nil {
 		return err
 	}
 
-	if unchanged {
-		var buf bytes.Buffer
-		if err = s.codec.Encode(updatedObj, &buf); err != nil {
-			return err
-		}
-		if _, err := s.convertToObject(buf.Bytes(), destination); err != nil {
-			return err
-		}
-		return nil
-	}
-
-	var (
-		value []byte
-		rv    int64
-	)
-	if created {
-		value, err = s.prepareObjectForStorage(ctx, updatedObj)
-		if err != nil {
-			return err
-		}
-		rsp2, err := s.store.Create(ctx, &resource.CreateRequest{
-			Key:   req.Key,
-			Value: value,
-		})
+	var rv uint64
+	// Only update (for real) if the bytes have changed
+	if !bytes.Equal(req.Value, existingBytes) {
```

Comment:
this avoids a few `json.Encode()` cycles -- because it compares the input to the proposed output directly (no longer a function that would do it again)

Reply:
Seerms that prepareObjectForUpdate adds annotations ... meaning the bytes will always be different ? 
https://github.com/grafana/grafana/blob/a9d81ba605c20112325495345c85de5d3a0bb775/pkg/storage/unified/apistore/prepare.go#L134-L136

Reply:
yes... but not with https://github.com/grafana/grafana/pull/102260